Scottish-Tech-Army/Volunteer-app
# SVA-382 | Fix API tunnelling bug on some Windows machines

- [Description](#description)
  - [Type of change](#type-of-change)
- [Testing locally](#testing-locally)
- [Checklist](#checklist)

# Description

- @vglampard  has been having problems getting the API tunnelling command to work on Windows.  This could affect other devs too and we want to make getting set up as simple and pain-free as possible.
- Updated the API tunnelling command to probably be more reliable generally, by using `cross-var` and `dotenv-cli` packages, so we can have a single command that works across different machines.

Fixes # SVA-382

## Type of change

Please delete options that are not relevant


# Testing locally

@vglampard has already tried this out on her Windows machine and I have on Linux - be great to get it tested on at least one other Windows machine, and on a Mac

- Pull this branch then in your `api` directory do `npm install`
- Make sure you have `API_TUNNEL_SUBDOMAIN` set in your `api/.env` file (see the README if needed)
- In one terminal window, in your `api` directory, run `npm start` like normal
- In a second terminal window, in your `api` directory, run `npm run tunnel`  (note the name of the command has changed). If that works, it should tell you something similar to `your url is: https://nadia-bloggs-abc123etc.loca.lt`
- Try that URL in a web browser with `/v1/projects` on the end (similar to `https://nadia-bloggs-abc123etc.loca.lt/v1/projects`)

# Checklist

- [] I have commented my code, particularly in hard-to-understand areas
- [] I have removed any unnecessary comments or console logging
- [] I have addressed accessibility, if needed
- [] I have checked the app in dark mode, if making front-end design changes
- [] I have added tests that prove my fix is effective or that my feature works
- [] I have updated the version numbers in `package.json` files in the [app](DEPLOYMENT.md#app-deployment) and/or [api](DEPLOYMENT.md#api-deployment-on-aws) directories as needed
